Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -1559,7 +1604,19 @@ void TextEditor::recompute_visual_lines(size_t line_index)
     if (is_wrapping_enabled())
         visual_data.visual_rect = { m_horizontal_content_padding, 0, available_width, static_cast<int>(visual_data.visual_line_breaks.size()) * line_height() };
     else
-        visual_data.visual_rect = { m_horizontal_content_padding, 0, font().width(line.view()), line_height() };
+        visual_data.visual_rect = { m_horizontal_content_padding, 0, width_of_view(line.view()), line_height() };
+}
+
+// Width of a Utf32View, adjusted with the appropriate tab size
+int TextEditor::width_of_view(Utf32View visual_line_view) const
+{
+    auto num_tabs = 0;
+    for (auto code_point : visual_line_view) {
+        if (code_point == '\t')
+            num_tabs++;
+    }
+
+    return font().width(visual_line_view) - num_tabs * font().glyph_width('\t') + font().glyph_spacing() + num_tabs * font().glyph_width(' ') * hard_tab_width();
 }
```

Comment:
This algorithm seems insufficient. Since tabs cause realignment, strings like `(tab)`, `(space)(tab)` and `(space)(space)(tab)` should all have the same width.